CMakeLists.txt

```diff
@@ -77,6 +74,9 @@ option(use_prov_client "Enable provisioning client" OFF)
 option(use_tpm_simulator "tpm simulator type of hsm used with the provisioning client" OFF)
 option(use_edge_modules "Enable support for running modules against Azure IoT Edge" OFF)
 option(use_custom_heap "use externally defined heap functions instead of the malloc family" OFF)
+set(compileOption_C "" CACHE STRING "passes a string to the command line of the C compiler")
```

Comment:
Be noted that looks like we don't have any build scripts that actually test this option. Please reply if that's not correct.